# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/SymbolType.h
# Commit: 87ae5e3b6962
# Full Hash: 87ae5e3b6962b5db12bcb271a1fd2aef4e20a809
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2019-04-09 16:24:26
# Regressor Bug: 1540719
# File Overlap Count: 1
# Description:
#   Bug 1540719 - Perform a last ditch GC if symbol allocation fails as we do for most other GC things r=sfink
# ==============================================================================

diff -r 1b7309612d60 -r 87ae5e3b6962 js/src/vm/SymbolType.h
--- a/js/src/vm/SymbolType.h	Tue Apr 09 09:53:57 2019 +0300
+++ b/js/src/vm/SymbolType.h	Mon Apr 01 18:36:41 2019 +0100
@@ -49,7 +49,7 @@
   void operator=(const Symbol&) = delete;
 
   static Symbol* newInternal(JSContext* cx, SymbolCode code,
-                             js::HashNumber hash, JSAtom* description);
+                             js::HashNumber hash, js::HandleAtom description);
 
   static void staticAsserts() {
     static_assert(uint32_t(SymbolCode::WellKnownAPILimit) ==
@@ -62,7 +62,8 @@
   }
 
  public:
-  static Symbol* new_(JSContext* cx, SymbolCode code, JSString* description);
+  static Symbol* new_(JSContext* cx, SymbolCode code,
+                      js::HandleString description);
   static Symbol* for_(JSContext* cx, js::HandleString description);
 
   JSAtom* description() const { return description_; }
